### PR TITLE
Laura

### PR DESCRIPTION
**Pull request by lauraherman22**

### CODE REVIEW

**Review by stevendleung — COMMENTED** [review]
Thanks!

**Review by stevendleung — APPROVED** [review]

**Review by stevendleung — APPROVED** [review]

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 1]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
 
 # COMMAND ----------
 
+# Index label
+labelIndexer = StringIndexer(inputCol="dep_del15", outputCol="label")
+
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]: The indexer should only be fit on the train set.
```suggestion
labelIndexer.fit(train_GBT)
```

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 2]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
...
+train_GBT = labelIndexer.fit(train_GBT).transform(train_GBT)
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]:
```suggestion
train_GBT = labelIndexer.transform(train_GBT)
```

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 3]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
...
+validation_GBT = labelIndexer.fit(validation_GBT).transform(validation_GBT)
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]:
```suggestion
validation_GBT = labelIndexer.transform(validation_GBT)
```

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 4]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
...
+test_GBT = labelIndexer.fit(test_GBT).transform(test_GBT)
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]:
```suggestion
test_GBT = labelIndexer.transform(test_GBT)
```

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 5]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
...
+validation_GBT = stringIndexer.fit(validation_GBT).transform(validation_GBT)
+test_GBT = stringIndexer.fit(test_GBT).transform(test_GBT)
+
+# COMMAND ----------
+
+# Assemble categorical and numeric features into vector
+numeric = ["origin_num_flights","origin_avg_dep_delay", "origin_pct_dep_del15", "origin_avg_taxi_time", "origin_avg_weather_delay", "origin_avg_nas_delay", "origin_avg_security_delay", "origin_avg_late_aircraft_delay", "dest_num_flights","dest_avg_dep_delay", "dest_pct_dep_del15", "dest_avg_taxi_time", "dest_avg_weather_delay", "dest_avg_nas_delay", "dest_avg_security_delay", "dest_avg_late_aircraft_delay", "carrier_num_flights", "carrier_avg_dep_delay", "carrier_avg_carrier_delay", "origin_WND_speed_rate", "origin_CIG_ceiling_height", "origin_VIS_distance", "origin_TMP_air_temperature", "origin_DEW_dew_point_temp", "dest_WND_speed_rate", "dest_CIG_ceiling_height", "dest_VIS_distance", "dest_TMP_air_temperature", "dest_DEW_dew_point_temp"]
+
+features = categorical_index + numeric
+assembler = VectorAssembler(inputCols=features, outputCol="features")
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py:704]: I've never used the VectorAssembler, does it need to be fit before transform?

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 6]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
...
+validation_GBT = labelIndexer.fit(validation_GBT).transform(validation_GBT)
+test_GBT = labelIndexer.fit(test_GBT).transform(test_GBT)
+
+# Index features
+categorical = ["month", "day_of_week", "op_unique_carrier", "Holiday", "PREVIOUS_FLIGHT_DELAYED_FOR_MODELS", "origin_WND_type_code", "origin_CIG_ceiling_visibility_okay", "origin_VIS_variability", "dest_WND_type_code", "dest_CIG_ceiling_visibility_okay", "dest_VIS_variability", "crs_dep_hour"]
+
+categorical_index = [i + "_Index" for i in categorical]
+  
+stringIndexer = StringIndexer(inputCols=categorical, outputCols=categorical_index)
+
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]: Same comment as above, only fit on train data
```suggestion
    stringIndexer.fit(train_GBT)
```

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 7]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
...
+test_GBT = labelIndexer.fit(test_GBT).transform(test_GBT)
+
+# Index features
+categorical = ["month", "day_of_week", "op_unique_carrier", "Holiday", "PREVIOUS_FLIGHT_DELAYED_FOR_MODELS", "origin_WND_type_code", "origin_CIG_ceiling_visibility_okay", "origin_VIS_variability", "dest_WND_type_code", "dest_CIG_ceiling_visibility_okay", "dest_VIS_variability", "crs_dep_hour"]
+
+categorical_index = [i + "_Index" for i in categorical]
+  
+stringIndexer = StringIndexer(inputCols=categorical, outputCols=categorical_index)
+    
+train_GBT = stringIndexer.fit(train_GBT).transform(train_GBT)
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]:
```suggestion
train_GBT = stringIndexer.transform(train_GBT)
```

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 8]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
...
+validation_GBT = stringIndexer.fit(validation_GBT).transform(validation_GBT)
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]:
```suggestion
validation_GBT = stringIndexer.transform(validation_GBT)
```

File: notebooks/Airline Delays - Data Processing Pipeline.py [hunk 9]
```diff
@@ -658,6 +677,38 @@ def chain_delay_analysis (dep_time_diff_one_flight_before, dep_time_diff_two_fli
...
+test_GBT = stringIndexer.fit(test_GBT).transform(test_GBT)
```
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]:
```suggestion
test_GBT = stringIndexer.transform(test_GBT)
```

**Review by RLashofRegas — CHANGES_REQUESTED** [review]
We should only be fitting on train data and then transforming the other datasets with the fitted transformations.

File: notebooks/Airline Delays - Data Processing Pipeline.py (hunk 3 above)
Comment (RLashofRegas) [inline notebooks/Airline Delays - Data Processing Pipeline.py]: train_GBT, validation_GBT, and test_GBT are never defined.

**Review by RLashofRegas — CHANGES_REQUESTED** [review]

File: notebooks/Airline Delays - Data Processing Pipeline.py (hunk 5 above)
Comment (lauraherman22) [reply notebooks/Airline Delays - Data Processing Pipeline.py:704]: No it should just be able to transform

File: notebooks/Airline Delays - Data Processing Pipeline.py (hunk 3 above)
Comment (lauraherman22) [reply notebooks/Airline Delays - Data Processing Pipeline.py]: Fixed in next commit

**Review by RLashofRegas — APPROVED** [review]

**Pull request merged**